modules/tools/lsp/config.el

```
@@ -138,27 +142,10 @@ This also logs the resolved project root, if found, so we know where we are."
         ;; errors flash briefly and then disappear).
         lsp-ui-sideline-show-hover nil)
 
-  (set-lookup-handlers! 'lsp-ui-mode :async t
-    :definition 'lsp-ui-peek-find-definitions
-    :references 'lsp-ui-peek-find-references))
-
-
-(use-package! company-lsp
-  :when (featurep! :completion company)
-  :defer t
-  :init
-  ;; Make sure that `company-capf' is disabled since it is incompatible with
-  ;; `company-lsp' (see lsp-mode#884)
-  (add-hook! 'lsp-mode-hook
-    (defun +lsp-init-company-h ()
-      (if (not (bound-and-true-p company-mode))
-          (add-hook 'company-mode-hook #'+lsp-init-company-h t t)
-        (setq-local company-backends
-                    (cons +lsp-company-backend
-                          (remq 'company-capf company-backends)))
-        (remove-hook 'company-mode-hook #'+lsp-init-company-h t))))
-  :config
-  (setq company-lsp-cache-candidates 'auto)) ;; cache candidates for better performance
+  (when (featurep! +peek)
```

Please document this new flag in modules/tools/lsp/README.org (under Description > Module Flags) and in docs/modules.org (see other modules in that list with flags listed next to them).

---

should I also change it to something like `+no-peek` or `+xref`, so that users don't suddenly see a different frontend when updating doom-emacs without modifying their local config?

---

I think `+peek` is best. Disabling lsp-ui's lookup handlers does not guarantee it'll fall back to xref, so `+xref` is misleading.

Side note: flags that disable functionality should be prefixed with a minus, instead. e.g. `-peek`

---

ok, I'll change it back to `+peek` then

---

done